tests/base/src/test/java/org/wildfly/security/ssl/SSLv2HelloAuthenticationTest.java

```diff
@@ -0,0 +1,444 @@
+                "wildfly-ssl-test-config-v1_6.xml",
+                enabledClientProtocols,
+                enabledServerProtocols,
+                "TLSv1"); // We expect the negotiated protocol to be the highest version protocol in common
+        assertNotNull(identity);
+        assertEquals("Principal Name", "ladybird", identity.getPrincipal().getName());
+    }
+
+    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, String clientConfigFileName, String[] enabledProtocols, String negotiatedProtocol) throws Exception {
+        return performConnectionTest(serverContext, clientUri, clientConfigFileName, enabledProtocols, enabledProtocols, negotiatedProtocol);
+    }
+
+    private SecurityIdentity performConnectionTest(SSLContext serverContext, String clientUri, String clientConfigFileName, String[] enabledClientProtocols, String[] enabledServerProtocols,  String negotiatedProtocol) throws Exception {
+        System.setProperty("wildfly.config.url", SSLAuthenticationTest.class.getResource(clientConfigFileName).toExternalForm());
+        AccessController.doPrivileged((PrivilegedAction<Integer>) () -> Security.insertProviderAt(new WildFlyElytronProvider(), 1));
+
+        AuthenticationContext context = AuthenticationContext.getContextManager().get();
+        AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
+        SSLContext clientContext = contextConfigurationClient.getSSLContext(URI.create(clientUri), context);
+
+        SSLServerSocketFactory sslServerSocketFactory = serverContext.getServerSocketFactory();
+
+        SSLServerSocket sslServerSocket = (SSLServerSocket) sslServerSocketFactory.createServerSocket(1111, 10, InetAddress.getLoopbackAddress());
+
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        Future<SSLSocket> socketFuture = executorService.submit(() -> {
+            try {
+                System.out.println("About to connect client");
+                SSLSocket sslSocket = (SSLSocket) clientContext.getSocketFactory().createSocket(InetAddress.getLoopbackAddress(), 1111);
+                sslSocket.getSession();
+
+                return sslSocket;
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            } finally {
+                System.out.println("Client connected");
+            }
+        });
+
+        SSLSocket serverSocket = (SSLSocket) sslServerSocket.accept();
+        SSLSession serverSession = serverSocket.getSession();
+        SSLSocket clientSocket = socketFuture.get();
+        SSLSession clientSession = clientSocket.getSession();
+
+        try {
+            // Ensure the enabled protocols are the only ones that were configured
+            Set<String> serverProtocols = new HashSet<>(Arrays.asList(serverSocket.getEnabledProtocols()));
```

Comment:
Instead of converting to a `Set`, I think you can use `assertArrayEquals(enabledServerProtocols,serverSocket.getEnabledProtocols())`.

Reply:
I think this function compares that the contents are the same including the order which causes failures depending on the JDK